cli/tests/integration/vendor_tests.rs

```diff
@@ -0,0 +1,331 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::serde_json;
+use deno_core::serde_json::json;
+use pretty_assertions::assert_eq;
+use std::fs;
+use std::process::Stdio;
+use tempfile::TempDir;
+use test_util as util;
+use util::http_server;
+
+#[test]
+fn output_dir_exists() {
+  let t = TempDir::new().unwrap();
+  let vendor_dir = t.path().join("vendor");
+  fs::write(t.path().join("mod.ts"), "").unwrap();
+  fs::create_dir_all(&vendor_dir).unwrap();
+  fs::write(vendor_dir.join("mod.ts"), "").unwrap();
+
+  let deno = util::deno_cmd()
+    .current_dir(t.path())
+    .env("NO_COLOR", "1")
+    .arg("vendor")
+    .arg("mod.ts")
+    .stderr(Stdio::piped())
+    .spawn()
+    .unwrap();
+  let output = deno.wait_with_output().unwrap();
+  assert_eq!(
+    String::from_utf8_lossy(&output.stderr).trim(),
+    format!(
+      concat!(
+        "error: Directory {} was not empty. Please provide an empty ",
+        "directory or use --force to ignore this error and potentially ",
+        "overwrite its contents.",
+      ),
+      vendor_dir.display(),
+    ),
+  );
+  assert!(!output.status.success());
+
+  // ensure it errors when using the `--output` arg too
+  let deno = util::deno_cmd()
+    .current_dir(t.path())
+    .env("NO_COLOR", "1")
+    .arg("vendor")
+    .arg("--output")
+    .arg("vendor")
+    .arg("mod.ts")
+    .stderr(Stdio::piped())
+    .spawn()
+    .unwrap();
+  let output = deno.wait_with_output().unwrap();
+  assert_eq!(
+    String::from_utf8_lossy(&output.stderr).trim(),
```

Comment:
Would be neat if we had a less verbose way of running the deno process and getting the output in the tests.